src/Appwrite/Auth/Validator/PersonalData.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Appwrite\Auth\Validator;
+
+/**
+ * Validates user password string against their personal data
+ */
+class PersonalData extends Password
+{
+    public function __construct(
+        protected ?string $userId = null,
+        protected ?string $email = null,
+        protected ?string $name = null,
+        protected ?string $phone = null,
+        protected bool $strict = false
+    ) {
+    }
+
+    /**
+     * Get Description.
+     *
+     * Returns validator description
+     *
+     * @return string
+     */
+    public function getDescription(): string
+    {
+        return 'Password must not include any personal data like your name, email, phone number, etc.';
+    }
+
+    /**
+     * Is valid.
+     *
+     * @param mixed $value
+     *
+     * @return bool
+     */
+    public function isValid($password): bool
+    {
+        if (!$this->strict) {
+            $password = strtolower($password);
+            $this->userId = strtolower($this->userId);
+            $this->email = strtolower($this->email);
+            $this->name = strtolower($this->name);
+            $this->phone = strtolower($this->phone);
+        }
+
+        if ($this->userId && strpos($password, $this->userId) !== false) {
+            return false;
+        }
+
+        if ($this->email && strpos($password, $this->email) !== false) {
+            return false;
+        }
+
+        if ($this->email && strpos($password, explode('@', $this->email)[0] ?? '') !== false) {
```

Comment:
I'm not sure explode is the best way to go here. Might be worth checking with the team. Email has a complex format.

Reply:
Agreed, to be on parity with the existing solutions, we are currently doing checks to ensure the password does not contain either the entire email or the `firstpart@example.com` of the email.

```
email: hello@appwrite.io 
password: *hello* // fail
password: *hello@appwrite.io* // fail
password: foobar // pass
```

The current code achieves this.  Do you want to test for additional cases?